langchain-core/src/prompts/structured.ts

```diff
@@ -8,6 +12,30 @@ import {
   ChatPromptTemplateInput,
 } from "./chat.js";
 
+function isWithStructuredOutput(
+  x: unknown
+  // eslint-disable-next-line @typescript-eslint/ban-types
+): x is {
+  withStructuredOutput: (...arg: unknown[]) => Runnable;
+} {
+  return (
+    typeof x === "object" &&
+    x != null &&
+    "withStructuredOutput" in x &&
+    typeof x.withStructuredOutput === "function"
+  );
+}
+
+function isRunnableBinding(x: unknown): x is RunnableBinding<unknown, unknown> {
+  return (
+    typeof x === "object" &&
+    x != null &&
+    "lc_id" in x &&
+    Array.isArray(x.lc_id) &&
+    x.lc_id.join("/") === "langchain_core/runnables/RunnableBinding"
```

Comment:
We could use `.getName()`, would be simpler?

Reply:
(synced offline: ideally we want to match the exact class)